pkg/cmd/infra/router/router.go

```diff
@@ -107,6 +111,7 @@ func (o *RouterSelection) Bind(flag *pflag.FlagSet) {
 	flag.MarkDeprecated("enable-ingress", "Ingress resources are now synchronized to routes automatically.")
 	flag.StringVar(&o.ListenAddr, "listen-addr", env("ROUTER_LISTEN_ADDR", ""), "The name of an interface to listen on to expose metrics and health checking. If not specified, will not listen. Overrides stats port.")
 	flag.BoolVar(&o.WatchEndpoints, "watch-endpoints", isTrue(env("ROUTER_WATCH_ENDPOINTS", "")), "Watch Endpoints instead of the EndpointSlice resource.")
+	flag.BoolVar(&o.AllowExternalCertificates, "allow-external-certificates", isTrue(env("ROUTER_ENABLE_EXTERNAL_CERTIFICATE", "")), "Enable RouteExternalCertificate feature-gate, disabled by default.")
```

Comment:
Agreed, that router doesn't know anything about this feature gate.

I've updated it to `Enable RouteSecretManager plugin and validation of external certificates` because the main purpose of this flag is to enable `RouteSecretManager` plugin along with validating external certificates.